lib/profiler/get_tick.h

```diff
@@ -67,6 +67,15 @@ get_tick(void)
     return regs;
 }
 
+#elif defined(__APPLE__)
+#include <time.h>
+
+static __inline hrtime_t
+get_tick(void)
+{
+    return clock_gettime_nsec_np (CLOCK_UPTIME_RAW);
```

Comment:
> mach_timebase_info is the historical api but in the case of the M1 indeed makes less sense since it's brand new anyway.

AFAICT, clock_gettime_nsec_np() is a relatively new API. Should we check for its availability (during ./configure) before using it here, to avoid breaking builds on Apple platforms without clock_gettime_nsec_np()?

Alternatively, perhaps we can use a C++11 API that is guaranteed by the standard to be available? Have you considered [high_resolution_clock](https://en.cppreference.com/w/cpp/chrono/high_resolution_clock) or its friends? FWIW, Squid is already using high_resolution_clock.

Reply:
Yes I m aware of the chrono's api, but it is only for Mac M1 ARM64's case here, otherwise old Mac X86 cases fall into the first category.

Reply:
Sorry, I do not know how to reconcile your response with my questions or concerns.

Perhaps you are implying that older Apple environments, where clock_gettime_nsec_np() may not available, will be handled by earlier `#if` and `#elif` clauses? I do not see how negating all those earlier conditions would guarantee clock_gettime_nsec_np() availability on an Apple platform.

Perhaps more importantly, the PR changes are clearly not specific to the "Mac M1 ARM64's case". That may be the case you are interested in, and there is nothing wrong with that, of course, but the the proposed code does not reflect your interest. Needless to say, the more generic you can make your changes, the better -- I am not implying that you should add more `#if` conditions! I am only saying that you cannot use the missing conditions as if they were there.

I am not familiar with Apple conventions, and there may be some Apply-specific logic at play here, but I doubt it is a good idea to leave our reliance on any such conventions implicit.

Reply:
I added some comments and made sure it s the ARM case...

Reply:
Thank you for clarifying your position/code.

In Squid project experience, the approach of guessing foo() availability by checking for the environment Bar where foo() should be available often leads to build failures. Sooner or later, Bar starts missing foo(). In most cases, it is much better to check whether foo() is available than to check whether the build happens in the environment Bar where foo() should be available.

You may be able to convince others to accept these (allegedly) "Mac M1 line from 2020"-specific changes. I am not going to block them just because they violate the above rule of thumb.

If you want _my_ vote, please switch from "Mac M1 line from 2020" checks (at compile time) to "clock_gettime_nsec_np is available" checks (at ./configure) time. Those checks would not even be specific to Apple AFAICT... If you need help implementing those, please ask!

BTW, you have [said](https://github.com/squid-cache/squid/pull/757/files#r557698383) that you are familiar with the C++ chrono API (which requires no checks!). Is there a good reason to avoid that checkless API here? I apologize if you have already answered that question, and I just failed to extract/grok that answer.

Reply:
Ok I ve switched to configure time check.